gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -489,7 +489,7 @@ protected static void applyToAllUnaryMethods(
       }
     }
 
-    public abstract <B extends StubSettings<B>> StubSettings<B> build() throws IOException;
+    public abstract SettingsT build() throws IOException;
```

Comment:
The change looks fine, but I'm not sure if it is technically speaking a breaking change or not? If somebody can confirm that it is binary nonbreaking, then LGTM. I.e. take gax with this update, compile it and pack in a jar (`./gradlew clean install`) and use it with any existing gapic client, which contains classes extending this abstract class. If everything still works without recompiling any of the binaries (i.e.gapic client does not need to be recompiled to work with updated gax).

Reply:
generics get striped from the binary, so its binary compatible. Not sure if its source compatible though